### PR TITLE
bunch of fixes as I hack on windows.

### PR DESCRIPTION
**Pull request by danderson**

### CODE REVIEW

File: ipn/ipnlocal/local.go [hunk 1]
```diff
@@ -371,51 +371,35 @@ func (b *LocalBackend) setClientStatus(st controlclient.Status) {
...
+func (b *LocalBackend) findExitNodeID(nm *netmap.NetworkMap) (prefsChanged bool) {
 	// If we have a desired IP on file, try to find the corresponding
 	// node.
-	if !b.prefs.ExitNodeIP.IsZero() {
-		// IP takes precedence over ID, so if both are set, clear ID.
-		if b.prefs.ExitNodeID != "" {
-			b.prefs.ExitNodeID = ""
-			prefsChanged = true
-		}
+	if b.prefs.ExitNodeIP.IsZero() {
```
Comment (bradfitz) [inline ipn/ipnlocal/local.go:379]: this is a data race. Or you need to rename this to `findExitNodeIDLocked` and document that `b.mu` must be held.

File: ipn/prefs.go [hunk 2]
```diff
@@ -168,6 +168,11 @@ func (p *Prefs) pretty(goos string) string {
 	if p.ShieldsUp {
 		sb.WriteString("shields=true ")
 	}
+	if !p.ExitNodeIP.IsZero() {
+		fmt.Fprintf(&sb, "exit=%v ", p.ExitNodeIP)
```
Comment (bradfitz) [inline ipn/prefs.go:172]: Update tests? There's already a big table of them.

File: wgengine/router/ifconfig_windows.go [hunk 3]
```diff
@@ -662,13 +662,26 @@ func getInterfaceRoutes(ifc *winipcfg.IPAdapterAddresses, family winipcfg.Addres
...
+	routes4, err := getInterfaceRoutes(ifc, windows.AF_INET)
 	if err != nil {
 		return err
 	}
 
-	got := make([]*winipcfg.RouteData, 0, len(routes))
-	for _, r := range routes {
+	routes6, err := getInterfaceRoutes(ifc, windows.AF_INET6)
+	if err != nil {
+		// TODO: what if v6 unavailable?
```
Comment (bradfitz) [inline wgengine/router/ifconfig_windows.go:672]: I don't think you want to return the error here. If IPv6 is off, I think this'll just fail and it'll break IPv4-only hosts, making syncRoutes return early here.

Comment (danderson) [reply wgengine/router/ifconfig_windows.go:672]: Everywhere else, we error out if v6 returns errors. Filing a bug to figure out if we need to work harder, or if windows always returns something even if users push the "turn IPv6 off" buttons.

**Pull request merged**